config/routes.rb

```diff
@@ -1,7 +1,7 @@
 Rails.application.routes.draw do
-  get "/auth/:provider/callback", to: "sessions#omniauth"
-  post "/auth/:provider/callback", to: "sessions#omniauth"
-  get "/auth/:provider", to: redirect { |params, request| "/auth/#{params[:provider]}" }, as: :auth_provider
+  devise_for :users, controllers: {
```

Comment:
```suggestion
  devise_for :users, only: [:sessions, :unlocks], controllers: {
```
E giới hạn router cho các tính năng lại nha

Reply:
Em xoá đi nó bị lỗi nên em phải để như vậy ạ